File: yetutil/src/main/java/net/yet/util/ToastUtil.java
package net.yet.util;

import android.content.Context;
import android.widget.Toast;

import net.yet.util.app.App;

public class ToastUtil {
	public static void show(String msg) {
		showLong(msg);
	}

	public static void show(int res) {
		showLong(res);
	}

	public static void showLong(String msg) {
		show(App.getContext(), msg, Toast.LENGTH_LONG);
	}

	public static void showLong(int res) {
		show(App.getContext(), res, Toast.LENGTH_LONG);
	}

	public static void showLong(Context context, String msg) {
		show(context, msg, Toast.LENGTH_LONG);
	}

	public static void showLong(Context context, int res) {
		show(context, res, Toast.LENGTH_LONG);
	}

	public static void showShort(String msg) {
		show(App.getContext(), msg, Toast.LENGTH_SHORT);
	}

	public static void showShort(int res) {
		show(App.getContext(), res, Toast.LENGTH_SHORT);
	}

	public static void showShort(Context context, String msg) {
		show(context, msg, Toast.LENGTH_SHORT);
	}

	public static void showShort(Context context, int res) {
		show(context, res, Toast.LENGTH_SHORT);
	}

	public static void show(Context context, int msg, int duration) {
		show(context, App.resString(msg), duration);
	}

	public static void show(final Context context, final String msg, final int duration) {
		TaskUtil.fore(new Runnable() {

			@Override
			public void run() {
				Toast.makeText(context, msg, duration).show();
			}
		});
	}
}

File: yetutil/src/main/java/yet/util/ToastUtil.kt
package yet.util

import android.content.Context
import android.widget.Toast

import yet.ui.res.Res
import yet.util.app.App

object ToastUtil {
	fun show(msg: String) {
		showLong(msg)
	}

	fun show(res: Int) {
		showLong(res)
	}

	fun showLong(msg: String) {
		show(App.app, msg, Toast.LENGTH_LONG)
	}

	fun showLong(res: Int) {
		show(App.app, res, Toast.LENGTH_LONG)
	}

	fun showLong(context: Context, msg: String) {
		show(context, msg, Toast.LENGTH_LONG)
	}

	fun showLong(context: Context, res: Int) {
		show(context, res, Toast.LENGTH_LONG)
	}

	fun showShort(msg: String) {
		show(App.app, msg, Toast.LENGTH_SHORT)
	}

	fun showShort(res: Int) {
		show(App.app, res, Toast.LENGTH_SHORT)
	}

	fun showShort(context: Context, msg: String) {
		show(context, msg, Toast.LENGTH_SHORT)
	}

	fun showShort(context: Context, res: Int) {
		show(context, res, Toast.LENGTH_SHORT)
	}

	fun show(context: Context, msg: Int, duration: Int) {
		show(context, Res.str(msg), duration)
	}

	fun show(context: Context, msg: String, duration: Int) {
		TaskUtil.fore(Runnable { Toast.makeText(context, msg, duration).show() })
	}
}
